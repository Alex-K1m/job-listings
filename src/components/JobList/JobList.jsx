import cn from 'classnames';
import { arrayOf, bool, func, shape } from 'prop-types';
import React from 'react';

import JobItem from '../JobItem';
import styles from './JobList.module.scss';

const JobList = ({ jobs, onTagClick, extraMargin }) => (
  <ul
    className={cn(styles.jobList, {
      [styles.extraMargin]: extraMargin,
    })}
  >
    {jobs.map((job) => (
      <li key={job.id}>
        <JobItem data={job} onTagClick={onTagClick} />
      </li>
    ))}
  </ul>
);

JobList.defaultProps = {
  extraMargin: true,
};

JobList.propTypes = {
  jobs: arrayOf(shape()).isRequired,
  onTagClick: func.isRequired,
  extraMargin: bool,
};

export default JobList;
