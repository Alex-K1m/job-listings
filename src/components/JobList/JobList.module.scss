@use '../breakpoints' as brp;

.jobList {
  display: grid;
  grid-gap: 1.6rem;
  margin-top: 3.2rem;

  @include brp.sm-up {
    grid-gap: 2.4rem;
    margin-top: 4rem;
  }
}

.extraMargin {
  @include brp.sm-up {
    margin-top: 7.6rem;
  }
}
