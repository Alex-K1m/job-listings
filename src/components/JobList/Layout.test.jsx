import React from 'react';
import renderer from 'react-test-renderer';

import data from '../../data.json';
import JobList from '.';

describe('JobList', () => {
  it('renders correctly', () => {
    const jobs = data.map((job) => {
      const { role, level, languages, tools } = job;
      return {
        ...job,
        tags: [role, level, ...languages, ...tools],
        logoUrl: 'https://example.com',
      };
    });

    const json = renderer
      .create(<JobList jobs={jobs} onTagClick={jest.fn()} />)
      .toJSON();

    expect(json).toMatchSnapshot();
  });
});
