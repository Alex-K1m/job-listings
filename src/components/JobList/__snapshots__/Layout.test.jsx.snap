// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobList renders correctly 1`] = `
<ul
  className="jobList extraMargin"
>
  <li>
    <div
      className="job jobHighlighted"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Photosnap
          </h2>
          <div
            className="chip"
          >
            <span
              className="label"
            >
              New!
            </span>
          </div>
          <div
            className="chip dark"
          >
            <span
              className="label"
            >
              Featured
            </span>
          </div>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Senior Frontend Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            1d ago
          </span>
          <span>
            Full Time
          </span>
          <span>
            USA only
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Frontend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Senior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              HTML
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              CSS
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job jobHighlighted"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Manage
          </h2>
          <div
            className="chip"
          >
            <span
              className="label"
            >
              New!
            </span>
          </div>
          <div
            className="chip dark"
          >
            <span
              className="label"
            >
              Featured
            </span>
          </div>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Fullstack Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            1d ago
          </span>
          <span>
            Part Time
          </span>
          <span>
            Remote
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Fullstack
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Midweight
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Python
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              React
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Account
          </h2>
          <div
            className="chip"
          >
            <span
              className="label"
            >
              New!
            </span>
          </div>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Junior Frontend Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            2d ago
          </span>
          <span>
            Part Time
          </span>
          <span>
            USA only
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Frontend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Junior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              React
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Sass
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            MyHome
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Junior Frontend Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            5d ago
          </span>
          <span>
            Contract
          </span>
          <span>
            USA only
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Frontend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Junior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              CSS
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Loop Studios
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Software Engineer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            1w ago
          </span>
          <span>
            Full Time
          </span>
          <span>
            Worldwide
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Fullstack
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Midweight
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Ruby
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Sass
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            FaceIt
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Junior Backend Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            2w ago
          </span>
          <span>
            Full Time
          </span>
          <span>
            UK only
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Backend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Junior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Ruby
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              RoR
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Shortly
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Junior Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            2w ago
          </span>
          <span>
            Full Time
          </span>
          <span>
            Worldwide
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Frontend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Junior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              HTML
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Sass
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Insure
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Junior Frontend Developer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            2w ago
          </span>
          <span>
            Full Time
          </span>
          <span>
            USA only
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Frontend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Junior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Vue
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Sass
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            Eyecam Co.
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Full Stack Engineer
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            3w ago
          </span>
          <span>
            Full Time
          </span>
          <span>
            Worldwide
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Fullstack
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Midweight
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Python
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Django
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="job"
    >
      <div
        className="logo"
      >
        <img
          alt="Photosnap logo"
          className="logo"
          src="https://example.com"
        />
      </div>
      <div
        className="body"
      >
        <div
          className="heading"
        >
          <h2>
            The Air Filter Company
          </h2>
        </div>
        <a
          className="position"
          href="#"
        >
          <h3>
            Front-end Dev
          </h3>
        </a>
        <div
          className="details"
        >
          <span>
            1mo ago
          </span>
          <span>
            Part Time
          </span>
          <span>
            Worldwide
          </span>
        </div>
      </div>
      <hr
        className="hr"
      />
      <ul
        className="tagList"
      >
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Frontend
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Junior
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              JavaScript
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              React
            </div>
          </div>
        </li>
        <li
          className="tagListItem"
        >
          <div
            className="wrapper"
          >
            <div
              className="tag"
            >
              Sass
            </div>
          </div>
        </li>
      </ul>
    </div>
  </li>
</ul>
`;
